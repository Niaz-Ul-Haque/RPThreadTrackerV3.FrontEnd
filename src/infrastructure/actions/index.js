// UI
export { toggleSidebar, TOGGLE_SIDEBAR } from './ui/toggleSidebar';
export { toggleNewsAside, TOGGLE_NEWS_ASIDE } from './ui/toggleNewsAside';
export { toggleHeaderDropdown, TOGGLE_HEADER_DROPDOWN } from './ui/toggleHeaderDropdown';
export { toggleMobileSidebar, TOGGLE_MOBILE_SIDEBAR } from './ui/toggleMobileSidebar';

// Threads
export { fetchThreads, FETCH_THREADS } from './threads/fetchThreads';
export { fetchedThreadsSuccess, FETCHED_THREADS_SUCCESS } from './threads/fetchedThreadsSuccess';

// User
export { fetchUser, FETCH_USER } from './user/fetchUser';
export { fetchedUserSuccess, FETCHED_USER_SUCCESS } from './user/fetchedUserSuccess';

// UserSettings
export { fetchUserSettings, FETCH_USER_SETTINGS } from './userSettings/fetchUserSettings';
export { fetchedUserSettingsSuccess, FETCHED_USER_SETTINGS_SUCCESS } from './userSettings/fetchedUserSettingsSuccess';
export { updateUserSettings, UPDATE_USER_SETTINGS } from './userSettings/updateUserSettings';
export { updatedUserSettingsSuccess, UPDATED_USER_SETTINGS_SUCCESS } from './userSettings/updatedUserSettingsSuccess';
export { setHasDashboardAtAGlanceHidden, SET_HAS_DASHBOARD_AT_A_GLANCE_HIDDEN } from './userSettings/setHasDashboardAtAGlanceHidden';

// News
export { fetchNews, FETCH_NEWS } from './news/fetchNews';
export { fetchedNewsSuccess, FETCHED_NEWS_SUCCESS } from './news/fetchedNewsSuccess';
