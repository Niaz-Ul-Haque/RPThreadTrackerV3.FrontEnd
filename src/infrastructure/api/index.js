export { default as isValidSlug } from './isValidSlug'; // eslint-disable-line import/prefer-default-export
