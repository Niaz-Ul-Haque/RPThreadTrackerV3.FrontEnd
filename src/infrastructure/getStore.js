import { createStore, applyMiddleware, compose } from 'redux';
import { createLogger } from 'redux-logger';
import { Iterable } from 'immutable'
import createSagaMiddleware from 'redux-saga';
import { getQuery } from '../utility'
import reducers from './reducers';
import { initSagas } from './initSagas'

const stateTransformer = (state) => {
	if (Iterable.isIterable(state)) return state.toJS();
	else return state;
};

const logger = createLogger({
	stateTransformer,
});

export const getStore = () => {
	const sagaMiddleware = createSagaMiddleware();
	const middleWares = [sagaMiddleware];
	if (getQuery()['logger']) { middleWares.push(logger) }
	const composables = [applyMiddleware(...middleWares)]
	const enhancer = compose(
		...composables
	);
	const store = createStore(
		reducers,
		undefined,
		enhancer
	);
	initSagas(sagaMiddleware);
	return store;
};
