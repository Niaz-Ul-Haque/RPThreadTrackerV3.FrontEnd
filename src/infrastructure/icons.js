import fontawesome from '@fortawesome/fontawesome';
import faBolt from '@fortawesome/fontawesome-free-solid/faBolt';
import faClock from '@fortawesome/fontawesome-free-solid/faClock';
import faCog from '@fortawesome/fontawesome-free-solid/faCog';
import faDollarSign from '@fortawesome/fontawesome-free-solid/faDollarSign';
import faDownload from '@fortawesome/fontawesome-free-solid/faDownload';
import faEdit from '@fortawesome/fontawesome-free-solid/faEdit';
import faEnvelope from '@fortawesome/fontawesome-free-solid/faEnvelope';
import faExternalLinkAlt from '@fortawesome/fontawesome-free-solid/faExternalLinkAlt';
import faEye from '@fortawesome/fontawesome-free-solid/faEye';
import faFilter from '@fortawesome/fontawesome-free-solid/faFilter';
import faInfoCircle from '@fortawesome/fontawesome-free-solid/faInfoCircle';
import faKey from '@fortawesome/fontawesome-free-solid/faKey';
import faLock from '@fortawesome/fontawesome-free-solid/faLock';
import faPlayCircle from '@fortawesome/fontawesome-free-solid/faPlayCircle';
import faPowerOff from '@fortawesome/fontawesome-free-solid/faPowerOff';
import faQuestionCircle from '@fortawesome/fontawesome-free-solid/faQuestionCircle';
import faRandom from '@fortawesome/fontawesome-free-solid/faRandom';
import faSearch from '@fortawesome/fontawesome-free-solid/faSearch';
import faTags from '@fortawesome/fontawesome-free-solid/faTags';
import faTimes from '@fortawesome/fontawesome-free-solid/faTimes';
import faTrashAlt from '@fortawesome/fontawesome-free-solid/faTrashAlt';
import faUndo from '@fortawesome/fontawesome-free-solid/faUndo';
import faUnlock from '@fortawesome/fontawesome-free-solid/faUnlock';
import faUsers from '@fortawesome/fontawesome-free-solid/faUsers';
import faUser from '@fortawesome/fontawesome-free-solid/faUser';

const icons = {
	init: () => {
		fontawesome.library.add(faBolt);
		fontawesome.library.add(faClock);
		fontawesome.library.add(faCog);
		fontawesome.library.add(faDollarSign);
		fontawesome.library.add(faDownload);
		fontawesome.library.add(faEdit);
		fontawesome.library.add(faEnvelope);
		fontawesome.library.add(faExternalLinkAlt);
		fontawesome.library.add(faEye);
		fontawesome.library.add(faFilter);
		fontawesome.library.add(faInfoCircle);
		fontawesome.library.add(faKey);
		fontawesome.library.add(faLock);
		fontawesome.library.add(faPlayCircle);
		fontawesome.library.add(faPowerOff);
		fontawesome.library.add(faQuestionCircle);
		fontawesome.library.add(faRandom);
		fontawesome.library.add(faSearch);
		fontawesome.library.add(faTags);
		fontawesome.library.add(faTimes);
		fontawesome.library.add(faTrashAlt);
		fontawesome.library.add(faUndo);
		fontawesome.library.add(faUnlock);
		fontawesome.library.add(faUsers);
		fontawesome.library.add(faUser);
	}
};
export default icons;
