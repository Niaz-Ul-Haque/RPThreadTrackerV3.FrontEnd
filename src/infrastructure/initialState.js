import { fromJS } from 'immutable'
// TODO... finalize value list
export const initialState = fromJS({
	currentUser: null,
	placeholder: 'test',
	ui: {
		isNewsAsideOpen: false,
		isSidebarOpen: true,
		isHeaderDropdownOpen: false,
		isMobileSidebarOpen: true
	}
});
