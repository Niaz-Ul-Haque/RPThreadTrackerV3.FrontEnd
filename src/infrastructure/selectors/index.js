// News
export { default as getNewsUnreadCount, markUnreadNews } from './news/getNews';

// Threads
export { default as getActiveThreadCharacters } from './threads/active/getActiveThreadCharacters';
export { default as getActiveThreadPartners } from './threads/active/getActiveThreadPartners';
export { default as getActiveThreadLastPosters } from './threads/active/getActiveThreadLastPosters';
export { default as getActiveThreadTags } from './threads/active/getActiveThreadTags';
export { default as getMyTurnFilteredThreads } from './threads/active/my-turn/getMyTurnFilteredThreads';
export { default as getTheirTurnFilteredThreads } from './threads/active/their-turn/getTheirTurnFilteredThreads';
export { default as getQueuedFilteredThreads } from './threads/active/queued/getQueuedFilteredThreads';
export { default as getMyTurnThreads } from './threads/active/my-turn/getMyTurnThreads';
export { default as getTheirTurnThreads } from './threads/active/their-turn/getTheirTurnThreads';
export { default as getQueuedThreads } from './threads/active/queued/getQueuedThreads';

export { default as getArchivedThreads } from './threads/archived/getArchivedThreads';
export { default as getArchivedThreadCharacters } from './threads/archived/getArchivedThreadCharacters';
export { default as getArchivedThreadPartners } from './threads/archived/getArchivedThreadPartners';
export { default as getArchivedThreadTags } from './threads/archived/getArchivedThreadTags';
export { default as getArchivedFilteredThreads } from './threads/archived/getArchivedFilteredThreads';

export { default as getRecentActivity } from './threads/getRecentActivity';
export { default as getThreadCountsByCharacter } from './threads/getThreadCountsByCharacter';
export { default as getPublicThreads } from './threads/getPublicThreads';

export { default as getCharactersSortedByIdentifier } from './characters/getCharactersSortedByIdentifier';

export { default as getTagsSortedByTagText } from './tags/getTagsSortedByTagText';

export { default as getIsLoadingIconVisible } from './ui/getIsLoadingIconVisible';
