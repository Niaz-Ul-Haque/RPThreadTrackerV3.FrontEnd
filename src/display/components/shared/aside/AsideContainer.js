import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchNews } from '../../../../state/news/actions';
import Aside from './Aside';

const propTypes = {
	news: PropTypes.shape({
		items: PropTypes.arrayOf(PropTypes.shape({
			id: PropTypes.number,
			userTitle: PropTypes.string.isRequired,
			lastPostDate: PropTypes.string.isRequired
		}))
	}).isRequired,
	dispatch: PropTypes.func.isRequired
};

function mapStateToProps(state) {
	const { news } = state;
	return { news };
}

class AsideContainer extends Component {
	componentDidMount() {
		const { dispatch } = this.props;
		if (!this.props.news.items || !this.props.news.items.length) {
			dispatch(fetchNews());
		}
	}

	render() {
		return (
			<Aside news={this.props.news.items} />
		);
	}
}

AsideContainer.propTypes = propTypes;

export default connect(mapStateToProps)(AsideContainer);
