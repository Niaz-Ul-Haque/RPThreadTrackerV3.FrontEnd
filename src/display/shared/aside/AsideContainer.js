// #region imports
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Aside from './Aside';
import { markUnreadNews } from '../../../infrastructure/selectors';
// #endregion imports

const propTypes = {
	news: PropTypes.arrayOf(PropTypes.shape({})).isRequired
};

function mapStateToProps(state) {
	const labeledNewsItems = markUnreadNews(state);
	return { news: labeledNewsItems };
}

const AsideContainer = (props) => {
	const { news } = props;
	return (<Aside news={news} />);
};

AsideContainer.propTypes = propTypes;
export default connect(mapStateToProps)(AsideContainer);
