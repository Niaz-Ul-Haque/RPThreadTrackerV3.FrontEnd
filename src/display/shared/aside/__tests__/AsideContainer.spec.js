// #region imports
import React from 'react';
import { shallowWithState } from '~/testhelpers/helpers.unit';
import * as selectors from '../../../../infrastructure/selectors'; //eslint-disable-line
import AsideContainer from '../AsideContainer';
// #endregion imports

// #region mocks
jest.mock('../../../../infrastructure/selectors', () => ({
	markUnreadNews: jest.fn()
}));
jest.mock('../Aside', () => 'Aside');
// #endregion mocks

const createTestProps = (propOverrides) => ({
	...propOverrides
});

const createTestState = (stateOverrides) => ({
	news: [],
	...stateOverrides
});

beforeEach(() => {
	jest.resetAllMocks();
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const state = createTestState();
			selectors.markUnreadNews.mockReturnValue([]);
			const jsx = <AsideContainer {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot with news', () => {
			const props = createTestProps();
			const state = createTestState();
			selectors.markUnreadNews.mockReturnValue([
				{
					postId: '12345',
					postTitle: 'Test Title'
				},
				{
					postId: '23456',
					postTitle: 'Test Title 2'
				},
				{
					postId: '34567',
					postTitle: 'Test Title 3'
				}
			]);
			const jsx = <AsideContainer {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
	});
	describe('content', () => {
		it('should pass news as prop', () => {
			const props = createTestProps();
			const state = createTestState();
			selectors.markUnreadNews.mockReturnValue([
				{
					postId: '12345',
					postTitle: 'Test Title'
				},
				{
					postId: '23456',
					postTitle: 'Test Title 2'
				},
				{
					postId: '34567',
					postTitle: 'Test Title 3'
				}
			]);
			const jsx = <AsideContainer {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element.props().news).toHaveLength(3);
		});
	});
});
