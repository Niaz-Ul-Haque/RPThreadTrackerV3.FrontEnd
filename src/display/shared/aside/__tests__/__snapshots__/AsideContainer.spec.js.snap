// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Aside
  news={Array []}
/>
`;

exports[`rendering snapshots should render valid snapshot with news 1`] = `
<Aside
  news={
    Array [
      Object {
        "postId": "12345",
        "postTitle": "Test Title",
      },
      Object {
        "postId": "23456",
        "postTitle": "Test Title 2",
      },
      Object {
        "postId": "34567",
        "postTitle": "Test Title 3",
      },
    ]
  }
/>
`;
